File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.py
n = int(input())

start_idx = 0
end_idx = 0
sum = 1
count = 0

while start_idx < n:
    if sum == n:
        count += 1
        sum -= start_idx + 1
        start_idx += 1

    elif sum < n:
        end_idx += 1
        sum += end_idx + 1
    else:
        sum -= start_idx + 1
        start_idx += 1

print(count)

File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.java
import java.util.Scanner;

public class Main{
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int[] nums = new int[n + 1];
	
		int count = 1;
		int start = 1;
		int end = 1;
		int sum = 1;
		while(end < n)
		{
			if(sum == n) {
				count += 1;
				end += 1;
				sum += end;
			}
			else if(sum < n) {
				end += 1;
				sum += end;
			}
			else if(sum > n) {
				sum -= start;
				start += 1;
			}
		}
		System.out.println(count);
	}
}